pkg/R/utils.R

```diff
@@ -419,3 +419,82 @@ cleanClosure <- function(func, checkedFuncs = new.env()) {
   }
   func
 }
+
+# Common function for implmentation of zipRDD() and cartesian().
+# param
+#   x An RDD.
+#   other An RDD.
+#   operation The specified operation, "zip" or "cartesian".
+#
+# return value
+#   The result RDD.
+doZipOrCartesian <- function(x, other, operation = c("zip", "cartesian")) {
```

Comment:
I think I also preferred https://github.com/amplab-extras/SparkR-pkg/commit/02363086977cbbbd2c159c81dac5d2e9617263b4 as a single longer function is hard to read.

A general comment is that I think it's good to follow @shivaram's purposed interface `mergePartitions(checkLengths = T, compactResults = F)` (something like this; doesn't have to be exact) instead of `mergePartitions(zip)`. The former focuses on the behaviors and is possible to be reused outside of zip/cartesian?